packages/dev/core/src/Animations/animation.ts

```diff
@@ -1002,7 +1020,7 @@ export class Animation {
                     case Animation.ANIMATIONLOOPMODE_YOYO:
                         return floatValue;
                     case Animation.ANIMATIONLOOPMODE_RELATIVE:
-                        return state.offsetValue * state.repeatCount + floatValue;
+                        return (state.offsetValue ?? 0) * state.repeatCount + floatValue;

```

Comment:
instead of making test (this and the rest), wouldn't it be better to set default values to the objects used as IAnimationState objects? There are two of those - `evaluateAnimationState` in animation.ts, and this._animationState in runtimeAnimation.ts .

Reply:
Also, it seems like IAnimationState should be an interface, but for some reason it is a class. I believe this can be changed.